sdk/go/pulumi/context.go

```diff
@@ -505,7 +520,13 @@ func (ctx *Context) RegisterResource(
 func (ctx *Context) RegisterComponentResource(
 	t, name string, resource ComponentResource, opts ...ResourceOption) error {
 
-	return ctx.RegisterResource(t, name, nil, resource, opts...)
+	return ctx.RegisterResource(t, name, nil /*props*/, resource, opts...)
+}
+
+func (ctx *Context) RegisterRemoteComponentResource(
```

Comment:
@pgavlin, what do you think of this new function? Or did you envision exposing this some other way?

Reply:
I think that what you have here is fine--this should really only be called from generated code.